tools/snap/build_remote.py

```diff
@@ -0,0 +1,151 @@
+            break
+
+        retry = retry - 1
+
+    return {target: workspace}
+
+
+def _extract_state(project, output, status):
+    match = re.match(r'^.*arch=(\w+)\s+state=([\w ]+).*$', output)
+    if match:
+        arch = match.group(1)
+        state = status[project]
+        state[arch] = match.group(2)
+
+        status[project] = state
+
+
+def _dump_status(archs, status, final=False):
+    while True:
+        if final:
+            print('Results for remote build finished at {0}'.format(datetime.datetime.now()))
+        else:
+            print('Remote build status at {0}'.format(datetime.datetime.now()))
+        print(' project                    {0}'.format(''.join('| {0}                       '.format(arch)
+                                                               for arch in archs)))
+        print('----------------------------{0}'.format('+-----------------------------' * len(archs)))
+        for project, states in sorted(status.items()):
+            print(' {0} {1}'.format(
+                project + ' ' * (25 - len(project)),
+                ''.join(' | {0}'.format(states[arch] + ' ' * (27 - len(states[arch]))) for arch in archs)))
+        print('----------------------------{0}'.format('+-----------------------------' * len(archs)))
+        print()
+
+        sys.stdout.flush()
+
+        if final:
+            break
+
+        time.sleep(10)
+
+
+def _dump_results(targets, archs, status, workspaces):
+    failures = False
+    for target in targets:
+        for arch in archs:
+            result = status[target][arch]
+
+            if result != 'Successfully built':
+                failures = True
+
+                with open(join(workspaces[target], '{0}_{1}.txt'.format(target, arch))) as file_h:
+                    build_output = file_h.read()
+
+                print('Output for failed build target={0} arch={1}'.format(target, arch))
+                print('-------------------------------------------')
+                print(build_output)
+                print('-------------------------------------------')
+                print()
+
+    if not failures:
+        print('All builds succeeded.')
+    else:
+        print('Some builds failed.')
+
+    return failures
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('targets', nargs='+', choices=['ALL', 'DNS_PLUGINS', 'certbot', *PLUGINS],
+                        help='the list of snaps to build')
+    parser.add_argument('--archs', nargs='+', choices=['amd64', 'arm64', 'armhf'], default='amd64',
+                        help='the architectures for which snaps are built')
+    args = parser.parse_args()
+
+    archs = set(args.archs)
+    targets = set(args.targets)
+
+    if 'ALL' in targets:
+        targets.remove('ALL')
+        targets.update(['certbot', 'DNS_PLUGINS'])
+
+    if 'DNS_PLUGINS' in targets:
+        targets.remove('DNS_PLUGINS')
+        targets.update(PLUGINS)
+
+    status = Manager().dict()
```

Comment:
Given that garbage collectors are working by watching if a given object has any remaining references in the stack during a garbage election cycle, I am pretty sure that it will. But since the official doc is using Manager() as a context manager, let's keep things standard.